tutorials/interactive/index.md

```diff
@@ -0,0 +1,781 @@
+
+The `toggleFavorite()` method, which is called when the IconButton is pressed,
+calls `setState`. Calling `setState` is critical, because this tells
+the framework that the widget’s state has changed and the widget
+should redraw. The `toggleFavorite` function swaps the UI between
+1) a star icon and the number ‘41’, and 2) a star_border icon and the number ‘40’.
+
+<!-- _code/lakes-interactive/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+class FavoriteWidgetState extends State<FavoriteWidget> {
+  [[highlight]]bool isFavorited = true;[[/highlight]]
+  [[highlight]]int favoriteCount = 41;[[/highlight]]
+
+  [[highlight]]void toggleFavorite()[[/highlight]] {
+    [[highlight]]setState(()[[/highlight]] {
+      // If the lake is currently favorited, unfavorite it.
+      if (isFavorited) {
+        favoriteCount--;
+        isFavorited = false;
+        // Otherwise, favorite it.
+      } else {
+        favoriteCount++;
+        isFavorited = true;
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        new Container(
+          padding: new EdgeInsets.all(0.0),
+          child: new IconButton(
+            [[highlight]]icon: (isFavorited[[/highlight]]
+                [[highlight]]? new Icon(Icons.star)[[/highlight]]
+                [[highlight]]: new Icon(Icons.star_border)),[[/highlight]]
+            color: Colors.red[500],
+            [[highlight]]onPressed: toggleFavorite,[[/highlight]]
+          ),
+        ),
+        new SizedBox(
+          width: 18.0,
+          child: new Container(
+            [[highlight]]child: new Text('$favoriteCount'),[[/highlight]]
+          ),
+        ),
+      ],
+    );
+  }
+}
+{% endprettify %}
+
+<a name="step-4"></a>
+### Step 4: Plug the stateful widget into the widget tree
+
+Add your custom stateful widget to the widget tree in the app's
+build method. First, locate the code that creates the Icon and Text, and delete it:
+
+<!-- _code/lakes/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+// ...
+[[strike]]new Icon([[/strike]]
+  [[strike]]Icons.star,[[/strike]]
+  [[strike]]color: Colors.red[500],[[/strike]]
+[[strike]]),[[/strike]]
+[[strike]]new Text('41')[[/strike]]
+// ...
+{% endprettify %}
+
+In the same location, create the stateful widget:
+
+<!-- _code/lakes-interactive/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    Widget titleSection = new Container(
+      // ...
+      child: new Row(
+        children: [
+          new Expanded(
+            child: new Column(
+              // ...
+          ),
+          [[highlight]]new FavoriteWidget()[[/highlight]],
+        ],
+      ),
+    );
+
+    return new MaterialApp(
+      // ...
+    );
+  }
+}
+{% endprettify %}
+
+<br>That's it! When you hot reload the app, the star icon should now respond to taps.
```

Comment:
Can we use CSS to create more space after code blocks? (instead of resorting to `<br>`)